zome.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2013-2017, The MetaCurrency Project (Eric Harris-Braun, Arthur Brock, et. al.)
+// Use of this source code is governed by GPLv3 found in the LICENSE file
+//---------------------------------------------------------------------------------------
+
+// Data integrity engine for distributed applications -- a validating monotonic
+//
+// Zome struct encapsulates logically related code, from "chromosome"
+
+package holochain
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+type Zome struct {
+	Name        string
+	Description string
+	Code        string // file name of DNA code
+	CodeHash    Hash
+	Entries     map[string]EntryDef
+	NucleusType string
+	Functions   map[string]FunctionDef
```

Comment:
Just to check, The h.Zomes are recursively serialized into the genesis entry and hashed (then signed), so the two maps here are serialized into the signed DNA. All the values at the next level are simple, maybe only strings even.